### PR TITLE
Bw 602

### PR DESCRIPTION
**Pull request by milansar**

### CODE REVIEW

File: projects/shared-library/src/lib/core/components/login/login.component.html [hunk 1]
```diff
@@ -6,7 +6,7 @@ <h1 mat-dialog-title>Choose Login Method</h1>
     <button class="flex-item" mat-raised-button mat-dialog-close color="primary" (click)="fbLogin()">facebook</button>
     <button class="flex-item" mat-raised-button mat-dialog-close color="primary" (click)="twitterLogin()">Twitter</button>
     <button class="flex-item" mat-raised-button mat-dialog-close color="primary" (click)="githubLogin()">Github</button>
-    <button class="flex-item" mat-raised-button mat-dialog-close color="primary" (click)="passwordLogin()">Email/Password</button>
+    <!-- <button class="flex-item" mat-raised-button mat-dialog-close color="primary" (click)="passwordLogin()">Email/Password</button> -->
```
Comment (milansar) [inline projects/shared-library/src/lib/core/components/login/login.component.html]: should this be commented?

File: package.json [hunk 2]
```diff
@@ -23,127 +23,128 @@
...
     "karma-cli": "2.0.0",
-    "karma-coverage-istanbul-reporter": "2.0.5",
+    "karma-coverage-istanbul-reporter": "2.1.0",
     "karma-jasmine": "2.0.1",
     "karma-jasmine-html-reporter": "1.4.2",
     "katex": "^0.10.2",
+    "lodash": "^4.17.11",
     "lodash.template": "^4.5.0",
     "mathquill": "^0.10.1-a",
-    "nativescript-dev-typescript": "^0.9.0",
```
Comment (milansar) [inline package.json]: nativescript-dev-typescript needs to be removed?

File: package.json [hunk 3]
```diff
@@ -23,127 +23,128 @@
...
-    "ngx-swiper-wrapper": "^7.2.1",
+    "ngx-swiper-wrapper": "^8.0.1",
     "reflect-metadata": "^0.1.13",
     "rxjs": "^6.5.2",
     "stream": "0.0.2",
-    "tns-core-modules": "^5.4.2",
-    "uglifyjs-webpack-plugin": "^1.3.0",
-    "zone.js": "^0.8.4"
+    "tns-core-modules": "^6.0.3",
+    "tns-platform-declarations": "^6.0.3",
```
Comment (milansar) [inline package.json]: this should move to dev dependency

File: projects/shared-library/src/lib/core/services/user.service.ts [hunk 4]
```diff
@@ -57,6 +57,7 @@ export class UserService {
                         return value;
                     } else {
                         value = {};
+                        value['userId'] = friendList[index];
```
Comment (milansar) [inline projects/shared-library/src/lib/core/services/user.service.ts]: should this be created using dot notation ie value.userId , access key notation should only be used when creating key dynamically

File: projects/shared-library/src/lib/core/services/user.service.ts [hunk 5]
```diff
@@ -57,6 +57,7 @@ export class UserService {
...
                         value.created_uid = friendList[index];
```
Comment (milansar) [inline projects/shared-library/src/lib/core/services/user.service.ts]: it appears that created_uid and user userid both are set with same value, is it desired?

File: projects/trivia-admin/src/app/bulk/components/bulk-summary/bulk-summary-question/bulk-summary-question.component.html [hunk 6]
```diff
@@ -43,7 +43,7 @@
     <ng-container cdkColumnDef="download">
       <mat-header-cell *cdkHeaderCellDef> Download File </mat-header-cell>
       <mat-cell class="col-status cursor bulk-summary-question-table-cell" *cdkCellDef="let row">
-        <a (click)="downloadFile(row)">Download</a>
+        <a (click)="downloadFile()">Download</a>
```
Comment (milansar) [inline projects/trivia-admin/src/app/bulk/components/bulk-summary/bulk-summary-question/bulk-summary-question.component.html]: is this change correct? how downloadfile function will have information about which file to download

File: projects/trivia/src/app/bulk/components/bulk-summary/bulk-summary-question/bulk-summary-question.component.html [hunk 7]
```diff
@@ -45,7 +45,7 @@
     <ng-container cdkColumnDef="download">
       <mat-header-cell *cdkHeaderCellDef> Download File </mat-header-cell>
       <mat-cell class="col-status cursor bulk-summary-question-table-cell" *cdkCellDef="let row">
-        <a (click)="downloadFile(row)">Download</a>
+        <a (click)="downloadFile()">Download</a>
```
Comment (milansar) [inline projects/trivia/src/app/bulk/components/bulk-summary/bulk-summary-question/bulk-summary-question.component.html]: see comment above related to download file

File: projects/shared-library/src/lib/core/components/login/login.component.html (hunk 1 above)
Comment (piya99) [reply projects/shared-library/src/lib/core/components/login/login.component.html]: Yes, It is not being used in login also function passwordLogin() is not defined anywhere.

File: package.json (hunk 2 above)
Comment (piya99) [reply package.json]: nativescript-dev-typescript is deprecated in favour of bundle workflow https://www.nativescript.org/blog/the-future-of-building-nativescript-apps

File: projects/shared-library/src/lib/core/services/user.service.ts (hunk 5 above)
Comment (piya99) [reply projects/shared-library/src/lib/core/services/user.service.ts]: Yes, I added userId as it is being used everywhere and not created_uid.

File: projects/trivia-admin/src/app/bulk/components/bulk-summary/bulk-summary-question/bulk-summary-question.component.html (hunk 6 above)
Comment (piya99) [reply projects/trivia-admin/src/app/bulk/components/bulk-summary/bulk-summary-question/bulk-summary-question.component.html]: It is for sample file download, the change is correct.

File: projects/trivia/src/app/bulk/components/bulk-summary/bulk-summary-question/bulk-summary-question.component.html (hunk 7 above)
Comment (piya99) [reply projects/trivia/src/app/bulk/components/bulk-summary/bulk-summary-question/bulk-summary-question.component.html]: It is for sample file download, the change is correct.

**Comment by milansar** [comment]
@anihalaney , PR is ready for your review

**Pull request merged**